src/Types/VersionedModuleSource.cs

```diff
@@ -100,9 +169,46 @@ versionedModuleSource.ModuleCase switch
 /// <param name="Source">Source code of module</param>
 public sealed record ModuleV0(byte[] Source) : VersionedModuleSource(Source)
 {
+    internal override uint GetVersion() => 0;
+
     internal static ModuleV0 From(Grpc.V2.VersionedModuleSource.Types.ModuleSourceV0 moduleSourceV0) =>
         new(moduleSourceV0.Value.ToByteArray());
 
+    /// <summary>
+    /// Creates a WASM-module from byte array.
+    /// </summary>
+    /// <param name="source">WASM-module as a byte array.</param>
+    /// <exception cref="ArgumentException">The length of the supplied module exceeds "MaxLength".</exception>
+    public static ModuleV0 From(byte[] source)
+    {
+        if (source.Length > MaxLength)
+        {
+            throw new ArgumentException(
+                $"Size of a data is not allowed to exceed {MaxLength} bytes."
+            );
+        }
+
+        return new ModuleV0(source.ToArray());
+    }
+
+    /// <summary>
+    /// Creates an instance from a hex encoded string.
+    /// </summary>
+    /// <param name="hexString">The WASM-module represented as a hex encoded string representing at most "MaxLength" bytes.</param>
+    /// <exception cref="ArgumentException">The supplied string is not a hex encoded WASM-module representing at most "MaxLength" bytes.</exception>
+    public static ModuleV0 FromHex(string hexString)
+    {
+        try
+        {
+            var value = Convert.FromHexString(hexString);
+            return From(value);
+        }
+        catch (Exception e)
```

Comment:
**Request change:** This catches the invalid length exception thrown by `From` and still provide the exception below about hex encoding.
Maybe you can just skip the try catch and let the exception from `Convert.FromHexString` pop further up.